From b2cbdaf910b7cd9d6a7c246b7d8b568dcbde1ade Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <icenowy@aosc.xyz>
Date: Fri, 1 Jul 2016 17:02:43 +0800
Subject: [PATCH 3/4] ARM: sun8i: add axp20x-battery node for axp22x

Signed-off-by: Icenowy Zheng <icenowy@aosc.xyz>
---
 arch/arm/boot/dts/axp22x.dtsi | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/arm/boot/dts/axp22x.dtsi b/arch/arm/boot/dts/axp22x.dtsi
index f05cbf6..4abcac5 100644
--- a/arch/arm/boot/dts/axp22x.dtsi
+++ b/arch/arm/boot/dts/axp22x.dtsi
@@ -150,4 +150,9 @@
 		compatible = "x-powers,axp202-usb-power-supply";
 		status = "disabled";
 	};
+
+	battery: battery {
+		compatible = "x-powers,axp202-battery";
+		status = "disabled";
+	};
 };
-- 
2.9.0

