--- a/arch/arm/boot/dts/sun8i-h3-orangepi-2.dts	2017-07-17 19:30:22.758475498 +0800
+++ b/arch/arm/boot/dts/sun8i-h3-orangepi-2.dts	2017-07-17 23:36:21.869864844 +0800
@@ -115,6 +115,10 @@
 	status = "okay";
 };
 
+&de {
+	status = "okay";
+};
+
 &ehci1 {
 	status = "okay";
 };
@@ -126,6 +130,10 @@
 	status = "okay";
 };
 
+&hdmi {
+	status = "okay";
+};
+
 &ir {
 	pinctrl-names = "default";
 	pinctrl-0 = <&ir_pins_a>;
@@ -189,6 +197,10 @@
 	status = "okay";
 };
 
+&tcon0 {
+	status = "okay";
+};
+
 &uart0 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&uart0_pins_a>;
