--- a/arch/arm/boot/dts/sun8i-h3-orangepi-plus.dts	2017-06-23 21:44:27.769070886 +0800
+++ b/arch/arm/boot/dts/sun8i-h3-orangepi-plus.dts	2017-06-23 21:41:25.345086748 +0800
@@ -76,6 +76,14 @@
 	status = "okay";
 };
 
+&codec {
+	allwinner,audio-routing =
+		"Line Out", "LINEOUT",
+		"MIC1", "Mic",
+		"Mic",  "MBIAS";
+	status = "okay";
+};
+
 &mmc2 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&mmc2_8bit_pins>;
